src/internal.c

```diff
@@ -1753,13 +1753,13 @@ void InitSuitesHashSigAlgo(Suites* suites, int haveECDSAsig, int haveRSAsig,
         #ifdef WC_RSA_PSS
             if (tls1_2) {
             #ifdef WOLFSSL_SHA512
-                if (keySz >= MIN_RSA_SHA512_PSS_BITS) {
+                if (keySz == 0 || keySz >= MIN_RSA_SHA512_PSS_BITS) {
```

Comment:
Why is a keySz of zero valid?

Reply:
InitSuitesHashSigAlgo is called by InitSuites, InitSuites is called by wolfSSL_SetTmpDH() and wolfSSL_set_connect_state() when the SSL is being used as a client. In either case, there may not be a client certificate. keySz is the size of the key in any available certificate. If no certificate is available, keySz is 0.

Reply:
I'm not sure keySz == 0 is the best way to partially determine client side.  What if the client has a key, the connection is NOT mutual auth, but because the client key is smaller than the server key the server rejects the connection because the client is limiting hash algos.  If we need to know the connection side here let's pass that in.

Reply:
The check is wrong in the first place - I shouldn't have done this.
We are generating a list of algorithms to advertise to the other side - what can we support with the other side's certs/keys - not what we can support with our certs/keys.
Removed check in latest commit.